### PR TITLE
Go: Improve QHelp for `go/unvalidated-url-redirection`.

### PR DESCRIPTION
**Pull request by max-schaefer**
The example showed a different (and better) fix from what the help claimed, but the suggestion also had a subtle bug that I fixed at the same time.

### CODE REVIEW

File: go/ql/src/Security/CWE-601/OpenUrlRedirectGood.go [hunk 1]
```diff
@@ -3,21 +3,26 @@ package main
...
+	"strings"
 )
 
-func serve() {
+func serve1() {
 	http.HandleFunc("/redir", func(w http.ResponseWriter, r *http.Request) {
 		r.ParseForm()
-		target, err := url.Parse(r.Form.Get("target"))
+		targetUrl := r.Form.Get("target")
+		// replace all backslashes with forward slashes before parsing the URL
```
Comment (atorralba) [inline go/ql/src/Security/CWE-601/OpenUrlRedirectGood.go:13]: I think the query doesn't really require this, the hostname check below is enough for the alert to go away. Would this cause confusion among users if they expect the alert to pop up when this replacement doesn't happen?

**Review by atorralba — COMMENTED** [review]
LGTM, with one comment.

File: go/ql/src/Security/CWE-601/OpenUrlRedirectGood.go (hunk 1 above)
Comment (max-schaefer) [reply go/ql/src/Security/CWE-601/OpenUrlRedirectGood.go:13]: It might, but I thought it better to show a solution that covers all bases.

**Review by subatoi — APPROVED** [review]
👍 from Docs

**Pull request merged**